MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+        area.region = region;
+        area.rank = areaDropDict[@"rank"];
+        [region.areaSet addObject:area];
+    }
+    // If it wasn't already, the area is now attached to the region
+    
+    // We should now consider adding this drop to the area.
+    // First, check if the area drop already exists, by way of checking the idDecimalString
+    NSString *idDecimalString = areaDropDict[@"_id"];
+    AreaDrop *drop = [self getUniqueAreaDropWithDecimalID:idDecimalString];
+    //If the drop already exists...
+    if (drop) {
+        // ... do nothing. (At least for now).
+        return;
+    } else {
+        // This else is not needed, but helps for clarity IMO.
+        // Hook up the drop.
+        NSEntityDescription *newDropDescription = [NSEntityDescription entityForName:@"AreaDrop" inManagedObjectContext:self.managedObjectContext];
+        drop = [[AreaDrop alloc] initWithEntity:newDropDescription insertIntoManagedObjectContext:self.managedObjectContext];
+        drop.idDecimalString = idDecimalString;
+        drop.method = areaDropDict[@"method"];
+        NSString *percent = areaDropDict[@"chance"];
+        drop.percentValue = percent.integerValue;
+        drop.quantity = areaDropDict[@"number"];
+        drop.rank = areaDropDict[@"rank"];
+        
+        // Hook up the area relationship.  We already ensured the area exists.
+        drop.area = area;
+        [area.dropSet addObject:drop];
+        
+        // Time to hook up the item relationship.  The item may or may not already exist.
+        NSString *itemName = areaDropDict[@"item"];
+        Item *item = (Item *)[self getUniqueEntityWithEntityName:@"Item" withNameAttribute:itemName];
+        if (!item) {
+            NSEntityDescription *newItemDescription = [NSEntityDescription entityForName:@"Item" inManagedObjectContext:self.managedObjectContext];
+            item = [[Item alloc] initWithEntity:newItemDescription insertIntoManagedObjectContext:self.managedObjectContext];
+            item.name = itemName;
+        }
+        [item.areaSourceSet addObject:drop];
+        drop.item = item;
+        
+        // The Drop should now exist, and be properly hooked up to the area, which exists and is hooked up to the region.
+    }
+}
+
+# pragma mark - Helper Methods
+
+- (NSDictionary *)loadDictionaryFromJsonFileNamed:(NSString *)fileName
+{
+    NSString *filePath = [[NSBundle mainBundle] pathForResource:fileName ofType:@"json"];
+    NSError *error;
+    NSString *fileContents = [NSString stringWithContentsOfFile:filePath encoding:NSUTF8StringEncoding error:&error];
+    if (error) {
+        NSLog(@"Error reading file: %@", error.localizedDescription);
+        return nil;
+    }
+    return (NSDictionary *)[NSJSONSerialization JSONObjectWithData:[fileContents dataUsingEncoding:NSUTF8StringEncoding]
+                                                           options:0
+                                                             error:nil];
```

Comment:
If you're not passing the address of an `NSError *`, you should be passing `NULL` rather than `nil`, but given that you have error handling already in place in this method, you should actually pass an error object in and check success before returning.